janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -870,8 +870,10 @@ public void clearStorage() throws BackendException {
             final ClusterState clusterState = zkStateReader.getClusterState();
             for (final String collection : clusterState.getCollectionsMap().keySet()) {
                 logger.debug("Clearing collection [{}] in Solr",collection);
-                final CollectionAdminRequest.Delete request = CollectionAdminRequest.deleteCollection(collection);
-                solrClient.request(request, collection);
+                // Collection is not dropped because it may have been created externally
+                final UpdateRequest deleteAll = newUpdateRequest();
+                deleteAll.deleteByQuery("*:*");
```

Comment:
I'm not a great user of Solr, but I think is not a good idea to use delete by query.
I think it's the same as Elasticsearch, on large index delete by query will take days to be perform (https://www.elastic.co/guide/en/elasticsearch/reference/current/_deleting_documents.html).

I think the problem is not when JanusGraph drop the collection, but when it can not re-create it.

For me this code will have big performance issue and it's a good practice when for example you reindexing to drop the collection.

Reply:
@davidclement90 I'm not a Solr user myself either. But the [docs](https://wiki.apache.org/solr/FAQ#How_can_I_delete_all_documents_from_my_index.3F) indicate this as the method to delete all documents so I wonder if the implementation is more efficient than Elasticsearch delete by query. The update here restores the original implementation removed in https://github.com/JanusGraph/janusgraph/pull/499/commits/8522787d6fabd08784105a5937aa83b66d9daa5a.

JanusGraph is able to create the collection but it assumes the same name for the Solr config set parameter (or uses a global config set). It doesn't support a per-collection config set parameter. Perhaps we can create another issue to add support for this? The other thing on this is that this only applies to Solr Cloud. With Solr HTTP, which might be the more common access mode, JanusGraph does not support clearing at all (deleting documents or dropping collection).

Reply:
Right, the original implementation is restored, so it's the same as the previous behavior.